webstorage/set.window.js

```diff
@@ -0,0 +1,63 @@
+["localStorage", "sessionStorage"].forEach(function(name) {
+    [9, "x"].forEach(function(key) {
+        test(function() {
+            var value = "value";
+
+            var storage = window[name];
+            storage.clear();
+
+            assert_equals(storage[key], undefined);
+            assert_equals(storage.getItem(key), null);
+            assert_equals(storage[key] = value, value);
+            assert_equals(storage[key], "value");
+            assert_equals(storage.getItem(key), "value");
+        }, "Setting property for key " + key + " on " + name);
+
+        test(function() {
+            var value = {
+                toString: function() { return "value"; }
+            };
+
+            var storage = window[name];
+            storage.clear();
+
+            assert_equals(storage[key], undefined);
+            assert_equals(storage.getItem(key), null);
+            assert_equals(storage[key] = value, value);
+            assert_equals(storage[key], "value");
+            assert_equals(storage.getItem(key), "value");
+        }, "Setting property with toString for key " + key + " on " + name);
+
+        test(function() {
+            Storage[key] = "proto";
```

Comment:
The capital S looks wrong; isn't this undefined? What am I missing?

Reply:
A `.prototype` I didn't write. Added, and then fixed all the bugs I missed.